### PR TITLE
Adds Inspector to Gem Repo Screen

### PR DESCRIPTION
**Pull request by AMZN-nggieber**
Inspector displays name, link to repo, summary, additional info, and included gems.

zdOCll1nDm

### CODE REVIEW

File: Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        else
+        {
+            m_addInfoTitleLabel->hide();
+            m_addInfoTextLabel->hide();
+
+            m_addInfoSpacer->changeSize(0, 0, QSizePolicy::Fixed, QSizePolicy::Fixed);
+        }
+
+        // Included Gems
+        m_includedGems->Update("Included Gems", "", m_model->GetIncludedGemNames(modelIndex));
```
Comment (AMZN-ScottR) [inline Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp:89]: Use `tr()` for display strings

File: Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    void GemRepoInspector::InitMainWidget()
+    {
+        // Repo name and url link
+        m_nameLabel = new QLabel();
+        m_nameLabel->setObjectName("gemRepoInspectorNameLabel");
+        m_mainLayout->addWidget(m_nameLabel);
+
+        m_repoLinkLabel = new LinkLabel(QString("Repo Url"), QString(""), 12, this);
```
Comment (AMZN-ScottR) [inline Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp:102]: Use `tr()` for display strings and don't think `QString()` is explicitly needed here

File: Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        QFrame* hLine = new QFrame();
+        hLine->setFrameShape(QFrame::HLine);
+        hLine->setObjectName("horizontalSeparatingLine");
+        m_mainLayout->addWidget(hLine);
+        m_mainLayout->addSpacing(10);
+
+        // Additional information
+        m_addInfoTitleLabel = new QLabel();
+        m_addInfoTitleLabel->setObjectName("gemRepoInspectorAddInfoTitleLabel");
+        m_addInfoTitleLabel->setText("Additional Information");
```
Comment (AMZN-ScottR) [inline Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp:125]: Use `tr()` for display strings

File: Code/Tools/ProjectManager/Source/GemRepo/GemRepoModel.cpp [hunk 4]
```diff
@@ -81,6 +98,45 @@ namespace O3DE::ProjectManager
...
+        for (const QString& gemPath : repoGemPaths)
+        {
+            AZ::Outcome<GemInfo> gemInfoResult = PythonBindingsInterface::Get()->GetGemInfo(gemPath);
+            if (gemInfoResult.IsSuccess())
+            {
+                allGemInfos.append(gemInfoResult.GetValue());
+            }
+            else
+            {
+                QMessageBox::critical(nullptr, tr("Gem Not Found"), QString("Cannot find info for gem %1.").arg(gemPath));
```
Comment (AMZN-ScottR) [inline Code/Tools/ProjectManager/Source/GemRepo/GemRepoModel.cpp:133]: `tr()` can be used on the format string as well

File: Code/Tools/ProjectManager/Source/LinkWidget.cpp [hunk 5]
```diff
@@ -33,7 +34,7 @@ namespace O3DE::ProjectManager
 
     void LinkLabel::enterEvent([[maybe_unused]] QEvent* event)
     {
-        setStyleSheet("font-size: 10px; color: #94D2FF; text-decoration: underline;");
+        setStyleSheet(QString("font-size: %1px; color: #94D2FF; text-decoration: underline;").arg(QString::number(m_fontSize)));
```
Comment (AMZN-ScottR) [inline Code/Tools/ProjectManager/Source/LinkWidget.cpp:37]: Call to `QString::number()` shouldn't be necessary here (or below)

File: Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp (hunk 1 above)
Comment (AMZN-nggieber) [reply Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp:89]: Shoot I spaced out on tr here thanks for catching it!

File: Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp (hunk 2 above)
Comment (AMZN-nggieber) [reply Code/Tools/ProjectManager/Source/GemRepo/GemRepoInspector.cpp:102]: The explicit QString() was actually required but changing to tr() will cover that as well.

**Review by AMZN-ScottR — APPROVED** [review]

**Review by AMZN-Phil — APPROVED** [review]

**Review by AMZN-alexpete — APPROVED** [review]

**Pull request merged**